mopidy/core/playback.py

```diff
@@ -14,21 +14,25 @@
 class PlaybackController(object):
     pykka_traversable = True
 
-    def __init__(self, backends, core):
+    def __init__(self, audio, backends, core):
+        # TODO: these should be internal
         self.backends = backends
         self.core = core
+        self._audio = audio
 
-        self._current_tl_track = None
         self._stream_title = None
         self._state = PlaybackState.STOPPED
 
-    def _get_backend(self):
-        # TODO: take in track instead
-        track = self.get_current_track()
-        if track is None:
+        self._current_tl_track = None
+        self._pending_tl_track = None
+
+        if self._audio:
+            self._audio.set_about_to_finish_callback(self._on_about_to_finish)
```

Comment:
Won't this cause the audio actor to run core's code and alter core state?

If you make the callback public, you can probably fix this by using `self.core.actor_ref.proxy().playback.on_about_to_finish` as the callback.